### PR TITLE
Add a script that counts lines of code.

### PR DESCRIPTION
**Pull request by bangerth**
I have an occasional need to count the lines of code in deal.II, and write a new script every time. This is a waste of time, and we have a place for such little scripts -- so here it is.

For reference, the output currently can be visualized to this:
line-count

### CODE REVIEW

**Comment by bangerth** [comment]
Yes, I know. I don't put much weight on the absolute number. The point is that we have had more or less continuous growth for 20 years :-)

**Comment by bangerth** [comment]
Ping?

**Comment by masterleinad** [comment]
Would you mind to run [`ShellCheck`](https://www.shellcheck.net/) on the script and go through the recommendations?

**Comment by bangerth** [comment]
I did as you suggested. There are two warnings left that are invalid and that I ignored.

File: contrib/utilities/count_lines.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# when this script was written, the repository had about 41,000 commits,
+# so counting lines for every 20th commit takes approximately 400 seconds,
+# or 6m30s.
+#
+# NOTE: The script has existed in this repository only for a finite time.
+#       So, if you go back in history, as this script does, it will
+#       eventually disappear. As a consequence, you cannot execute it
+#       in its current place -- copy it to the top level of the working
+#       copy you are working in, where it will be an untracked file that
+#       git leaves alone as it cycles through repository commits.
```
[the remark this answers is not in the document]
Comment (masterleinad) [reply contrib/utilities/count_lines.sh]: Fine with me.

**Pull request merged**